src/adapter/src/util.rs

```diff
@@ -36,6 +36,7 @@ use crate::{ExecuteResponse, PeekResponseUnary};
 pub struct ClientTransmitter<T: Transmittable> {
     tx: Option<oneshot::Sender<Response<T>>>,
     internal_cmd_tx: UnboundedSender<Message>,
+    metrics: Option<(Metrics, &'static str)>,
```

Comment:
I'm not convinced that putting this on ClientTransmitter is a good choice. `adapter_client` I think is a better fit.

Reply:
alright, i added it in both places - happy to remove that second commit adding it to the ClientTransmitter if you think it's just unnecessary duplication